src/game/Object/Player.cpp

```diff
@@ -2204,12 +2204,27 @@ void Player::SetGameMaster(bool on)
 
         GetHostileRefManager().setOnlineOfflineState(false);
         CombatStopWithPets();
+        
+        if (Pet* pet = GetPet())
+        {
+            if (m_ExtraFlags |= PLAYER_EXTRA_GM_ON)
```

Comment:
I meant this exact line `if (m_ExtraFlags |= PLAYER_EXTRA_GM_ON)`. Should be true as it will evaluate to at least PLAYER_EXTRA_GM_ON which is not 0
and |= is assignment, which is already done in the code above as well (not in the added part).